tests/test_main.py

```
@@ -81,6 +83,148 @@ def _mock_run_with_keyboard_interrupt(coro: Any) -> None:
+            OSError,
+        ) as exc:
+            future.set_exception(exc)
+        else:
+            future.set_result(result)
+        return future
+
+    def shutdown(self, wait: bool = False, cancel_futures: bool = False) -> None:
+        self.calls.append((wait, cancel_futures))
+        if self.shutdown_typeerror and cancel_futures is True:
+            # Simulate older Python versions that do not accept cancel_futures.
+            raise TypeError("cancel_futures unsupported")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)